app/src/main/java/com/safetyheads/akademiaandroida/token/FirebaseTokenRepository.kt

```diff
@@ -0,0 +1,14 @@
+package com.safetyheads.akademiaandroida.token
+
+import com.google.firebase.messaging.FirebaseMessaging
+import com.safetyheads.akademiaandroida.domain.repositories.TokenRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.tasks.await
+
+class FirebaseTokenRepository: TokenRepository {
+    override fun getMessagingToken(): Flow<String> = flow {
+        val token = FirebaseMessaging.getInstance().token.await()
+        emit(token)
```

Comment:
emit(FirebaseMessaging.getInstance().token.await()) wouldn't work?

Reply:
I would recommend making 'FirebaseMessaging.getInstance()' as Singleton in AndroidAcademyApplication